app/ante/evm/06_account_verification.go

```diff
@@ -39,7 +40,9 @@ func VerifyAccountBalance(
 		account = statedb.NewEmptyAccount()
 	}
 
-	if err := keeper.CheckSenderBalance(sdkmath.NewIntFromBigInt(account.Balance), txData); err != nil {
+	adjustedBalance := utils.Convert6To18Decimals(*account.Balance)
```

Comment:
The balance here is already 18 decimals no? because comes from the stateDB, that uses the evm keeper `GetBalance` that returns the balance in 18 decs

Reply:
It actually comes up as 18 decimals here because I've modified the `GetBalance` function to scale up the to 18 decimals so it shows up in MM correctly as it can't interpret another precision.